### ISSUE TITLE
AN is crashing on startup after upgrade. Can you update the example app with every new version?

### ISSUE THREAD

**Issue by aytunch** [AppLovin/AppLovin-MAX-Flutter#211]
### MAX Plugin Version

3.9.2

### Flutter Version

3.19.6

### Device/Platform Info

Android

### Current Behavior

We have google ads and applovin during our transition and AB test period.
AN crashes on startup after upgrading to the latest 3.9.2 version.
The correct gradle versions are not stated anywhere and we can not look at the example app.
Please state the latest SDK to be used in both gradle files in the changelog or in readme

### How to Reproduce

After I upgrade, we changed:

```
/android/app/build.gradle
from:
    implementation 'com.applovin:applovin-sdk:12.3.0'
    implementation 'com.applovin.mediation:google-adapter:22.6.0.0'
to:
    implementation 'com.applovin:applovin-sdk:+'
    implementation 'com.applovin.mediation:google-adapter:23.0.0.0'

/android/build.gradle
from:
        classpath 'com.google.gms:google-services:4.3.10'
to:
        classpath 'com.google.gms:google-services:4.3.14'

from:
  applovin_max: 3.8.0
to:
  applovin_max: 3.9.2
```

**Comment by alhiwatan** [comment]
@aytunch, you have two issues here.

- Crash on startup: This is a duplicate of #210. You must call AppLovinMAX. initialize() at most once. Otherwise, you might have a crash.
- Updating an example app : You should upgrade `applovin_max `with `flutter pub upgrade` instead of manually editing config files.   Also, you should not have `com.applovin:applovin-sdk:+` in your `build.gradle`.  The applovin_max package contains a specific version of `com.applovin:applovin-sdk` since each version of `applovin_max` can support its own version of `com.applovin:applovin-sdk`.   Specifying the latest may break the build or the behavior.

**Comment by ycv005** [comment]
@alhiwatan Earlier it was working. https://github.com/AppLovin/AppLovin-MAX-Flutter/issues/212#issuecomment-2109393850

**Comment by bigjjjim** [comment]
I understand now we must call the initialize() only once in the lifetime of the app but how do we do while developping? I am confused whether the solution is to actually not use the hot restart feature of Flutter or if there is a way to avoid calling the method specifically on hot restart. Any help is appreciated!

**Comment by alhiwatan** [comment]
@bigjjjim, we have just released [3.10.1](https://github.com/AppLovin/AppLovin-MAX-Flutter/releases/tag/release_3_10_1) that fixes the issue. Please feel free to reopen if you still have the issue.

**Comment by ycv005** [comment]
@alhiwatan But the documentation is still not updated.
<img width="807" alt="image" src="https://github.com/AppLovin/AppLovin-MAX-Flutter/assets/26734819/8669ad8d-6bbe-44b1-9dfc-fa8e16ef931e">

**Comment by alhiwatan** [comment]
@ycv005, thanks for pointing that out.   We will update the comment as well.
